nomad-aws/main.tf

```diff
@@ -29,10 +29,10 @@ module "nomad_tls" {
 locals {
   # Creates the Nomad Security Group(SG) list for the Instances.
   # Will include SSH SG if var.ssh_key is not null.
-  nomad_security_groups = compact(list(
+  nomad_security_groups = compact(tolist([
```

Comment:
Nit: I _think_ this can also just be

```
compact([
  aws_security_group.nomad_sg.id,
  var.ssh_key != null ? aws_security_group.ssh_sg[0].id : "",
])
```

Not 100% sure, but if that works it could be a little more clean.

Reply:
Yes, you're right - Terraform is very lax with their handling of lists vs tuples and mostly handles them interchangeably.